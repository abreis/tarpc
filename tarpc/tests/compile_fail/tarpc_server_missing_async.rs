#![allow(incomplete_features)]
#![feature(generic_associated_types)]

#[tarpc::service]
trait World {
    async fn hello(name: String) -> String;
}

struct HelloServer;

#[tarpc::server]
impl World for HelloServer {
    fn hello(name: String) ->  String {
        format!("Hello, {}!", name)
    }
}

fn main() {}
