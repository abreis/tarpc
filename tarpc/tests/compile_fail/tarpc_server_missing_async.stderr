error: not all trait items implemented, missing: `HelloFut`
  --> $DIR/tarpc_server_missing_async.rs:12:1
   |
12 | / impl World for HelloServer {
13 | |     fn hello(name: String) ->  String {
14 | |         format!("Hello, {}!", name)
15 | |     }
16 | | }
   | |_^

error: hint: `#[tarpc::server]` only rewrites async fns, and `fn hello` is not async
  --> $DIR/tarpc_server_missing_async.rs:13:5
   |
13 |     fn hello(name: String) ->  String {
   |     ^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type or module `serde`
 --> $DIR/tarpc_server_missing_async.rs:4:1
  |
4 | #[tarpc::service]
  | ^^^^^^^^^^^^^^^^^ use of undeclared type or module `serde`
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
